casparser_isin/mf_isin.py

```diff
@@ -81,24 +81,30 @@ def scheme_lookup(self, rta: str, scheme_name: str, rta_code: str):
 
         sql = """SELECT name, isin, amfi_code, type from scheme"""
         where = ["rta = :rta"]
-        args = {"rta": RTA_MAP.get(str(rta).upper(), "")}
 
-        if re.search("re-*invest", scheme_name, re.I):
-            where.append("name LIKE '%reinvest%'")
-        else:
-            where.append("name NOT LIKE '%reinvest%'")
+        if re.search(r"fti(\d+)", rta_code, re.I) and rta.upper() in ("CAMS", "FRANKLIN", "FTAMIL"):
+            # Try searching db for Franklin schemes
+            where.append("rta_code = :rta_code")
+            args = {"rta": "FRANKLIN", "rta_code": rta_code}
+            sql_statement = "{} WHERE {}".format(sql, " AND ".join(where))
+            results = self.run_query(sql_statement, args)
+            if len(results) != 0:
+                return results
 
-        if match := re.search(r"fti(\d+)", rta_code, re.I):
-            amc_code = match.group(1)
-            where.append("amc_code = :amc_code")
-            args.update(amc_code=amc_code)
+        args = {"rta": RTA_MAP.get(str(rta).upper(), "")}
+
+        if "hdfc" in scheme_name.lower() and re.search(r"^h\d+$", rta_code, re.I):
+            # Special case for old HDFC funds with scheme codes of format "H\d+"
+            if re.search("re-*invest", scheme_name, re.I):
+                where.append("name LIKE '%reinvest%'")
+            where.append("rta_code like :rta_code")
```

Comment:
This change breaks other scenarios. ex:

`"63"	"HDFC TAX SAVER - GROWTH OPTION"	"INF179K01BB8"	"101979"	"EQUITY"	"CAMS"	"H32"	"32"
"7329"	"HDFC TAX SAVER - DIRECT PLAN - GROWTH OPTION"	"INF179K01YS4"	"119060"	"EQUITY"	"CAMS"	"H32T"	"32T"
`

Reply:
Can you please post the exact string `"{RTA_CODE}-{SCHEME_NAME_DETAILED}"` from the CAS pdf for the breaking schemes?

Reply:
H32-HDFC TaxSaver - Regular Plan - Growth
H02-HDFC Flexi Cap Fund - Regular Plan - Growth

Reply:
Tweaked the check a little bit. Can you please test this again if you got some time?

Reply:
Seems to work now. One call out on franklin though, the amfi_code for "Franklin India Ultra Short Bond Fund- Super Inst Plan-Segregated Portfolio 1 **** " is missing

https://www.amfiindia.com/spages/NAVOpen.txt

Reply:
yeah. I am aware of it. The amfi code is populated based on the scheme ISIN and it looks like the segragated portfolio schemes in the NAV txt file don't have ISINs. 

working on manually mapping the amfi code for the missing schemes.

Reply:
Many thanks for the review. I'll add the AMFI codes for missing schemes soon.